xplat dnx on windows and some other minor changes

glennc opened the pull request:
@anurse I don't want to merge this right now. But it seems like a good time to get a code review for powershell-isms I might be missing. Also good to get a check on the runtimeInfo changes since it will become the template to get rid of GetArch, GetVersion, etc. Can you take a look?

glennc commented on src/dnvm.ps1:
```diff
@@ -866,9 +990,20 @@ function dnvm-list {
+            if($Detailed) {
+                $items | 
+                    Sort-Object Version, Runtime, Architecture, OperatingSystem, Alias | 
+                    Format-Table -AutoSize -Property @{name="Active";expression={if($_.Active) { "*" } else { "" }};alignment="center"}, "Version", "Runtime", "Architecture", "OperatingSystem", "Alias", "Location" | Out-String| ColorActive -color green
```
This is wrong, no need for -color fixed on my machine will push after more review feedback.

The effect of this is nice, active row colored blue. But I don't know if this is the most effective way of achieving it. Thoughts?

anurse commented on src/dnvm.ps1:
```diff
@@ -251,6 +254,64 @@ function Safe-Filecopy {
+        $runtimeInfo.OS = $DefaultOperatingSystem
+    }
+
+    if($runtimeInfo.OS -eq "osx") {
```
Comment about normalizing "osx" to "darwin"

glennc replied:
response about normalizing "osx" to "darwin"

anurse replied:
:trollface: I meant add a comment :P

anurse commented on src/dnvm.ps1:
```diff
@@ -251,6 +254,64 @@ function Safe-Filecopy {
+
+    if([String]::IsNullOrWhiteSpace($runtimeInfo.OS)) {
+        if($runtimeInfo.Runtime -eq "mono"){
+            #If OS is empty and you are asking for mono, i.e `dnvm install latest -os mono` then we don't know what OS to pick. It could be Linux or Darwin.
```
Wouldn't the example here be `dnvm install latest -r mono` ?

glennc replied:
Yes...

anurse commented on src/dnvm.ps1:
```diff
@@ -251,6 +254,64 @@ function Safe-Filecopy {
+        $runtimeInfo.OS = "darwin"
+    }
+
+    if([String]::IsNullOrWhiteSPace($runtimeInfo.Runtime)) {
```
TyPo

anurse commented on src/dnvm.ps1:
```diff
@@ -251,6 +254,64 @@ function Safe-Filecopy {
+        $runtimeInfo.Runtime = $OSRuntimeDefaults.Get_Item($runtimeInfo.OS)
+    }
+
+    if([String]::IsNullOrWhiteSPace($runtimeInfo.Architecture)) {
```
TyPo

anurse commented on src/dnvm.ps1:
```diff
@@ -1382,6 +1603,35 @@ function dnvm-setup {
+    
+    if (-not $runtimesInstall){
+        $title = "Getting started"
+        $message = "It looks like you don't have any runtimes installed. Do you want us to install a $RuntimeShortFriendlyName to get you started?"
```
This seems more complicated than it should be. I wonder if there's a powershell-native way to do this rather than calling into the host via .NET

anurse replied:
Looks like there isn't. So this seems ok.

anurse commented:
Looks good to me

glennc commented:
@anurse. Updated with feedback and continued some more of the refactor, removed methods that aren't needed anymore. I will wait until I've merged in the PSv2.0 changes before trying to merge, but it should be ready except for that.

dnfclas commented:
__@glennc__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

BrennanConroy commented:
This will fix #332

BrennanConroy commented:
ping @anurse

anurse commented:
:shipit:

anurse commented:
:shipit:
